adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -510,12 +560,30 @@ case class ReferenceRegion(
   }
 
   override def hashCode: Int = {
-    var result = 37
-    result = 41 * result + (if (referenceName != null) referenceName.hashCode else 0)
-    result = 41 * result + start.hashCode
-    result = 41 * result + end.hashCode
-    result = 41 * result + (if (strand != null) strand.ordinal() else 0)
-    result
+    // add 37 here because this is the start of the hashCode
+    val nameHashCode = 37 + {
```

Comment:
Case classes have an auto-genned hashCode. I don't think we should be implementing one here. What's the git blame for the hashCode function say?